# Intro data science - portfolio assignment 8 - Confidence intervals Spotify dataset

### Assignment:
### - Load the dataset from portfolio assignment 4
### - Calculate the 90%, 95%, 99% and 99.99% confidence interval for at least 2 columns with numerical data. Do you see the impact the confidence has on the interval?
### - Now calculate the 95% confidence interval again but use only the first 10% of your rows. Compare this interval to the previous 95% confidence interval you calculated. Do you see the impact of having less data?

In [1]:
import pandas as pd
import numpy as np
import scipy.stats as st

In [3]:
songs = pd.read_csv('datasets/04-top-spotify-songs-10s.csv', sep=',', engine='python')

In [4]:
songs.head()

Unnamed: 0                 title      artist        top genre  year  bpm  \
0           1      Hey, Soul Sister       Train       neo mellow  2010   97   
1           2  Love The Way You Lie      Eminem  detroit hip hop  2010   87   
2           3               TiK ToK       Kesha        dance pop  2010  120   
3           4           Bad Romance   Lady Gaga        dance pop  2010  119   
4           5  Just the Way You Are  Bruno Mars              pop  2010  109   

   nrgy  dnce  dB  live  val  dur  acous  spch  pop  
0    89    67  -4     8   80  217     19     4   83  
1    93    75  -5    52   64  263     24    23   82  
2    84    76  -3    29   71  200     10    14   80  
3    92    70  -4     8   71  295      0     4   79  
4    84    64  -5     9   43  221      2     4   78

For this demonstration I will be calculating the confidence intervals for the nrgy (energy) and dnce (danceability) columns.

Let's start with the energy level.

In [5]:
songs['nrgy'].mean()

70.50414593698176

The mean energy level of all songs is 70.5. What would the interval be with a certain confidence %?

In [6]:
confidence = 0.9

st.t.interval(confidence, len(songs)-1, loc=songs['nrgy'].mean(), scale=st.sem(songs['nrgy']))

(69.40991475566787, 71.59837711829564)

In [7]:
confidence = 0.95

st.t.interval(confidence, len(songs)-1, loc=songs['nrgy'].mean(), scale=st.sem(songs['nrgy']))

(69.19967287229983, 71.80861900166369)

In [8]:
confidence = 0.99

st.t.interval(confidence, len(songs)-1, loc=songs['nrgy'].mean(), scale=st.sem(songs['nrgy']))

(68.78778352638341, 72.22050834758011)

In [9]:
confidence = 0.999

st.t.interval(confidence, len(songs)-1, loc=songs['nrgy'].mean(), scale=st.sem(songs['nrgy']))

(68.30772356079197, 72.70056831317157)

Takeaway: As the confidence increases, so does the range. This is because the more confident we want to be about the mean energy level being correct, the higher the range should be. The less confident we have to be, the narrower the range gets, because there is a larger margin for error.

Let's do the exact same but this time with the danceability.

In [12]:
songs['dnce'].mean()

64.37976782752902

In [13]:
confidence = 0.9

st.t.interval(confidence, len(songs)-1, loc=songs['dnce'].mean(), scale=st.sem(songs['dnce']))

(63.48223172798374, 65.27730392707431)

In [14]:
confidence = 0.95

st.t.interval(confidence, len(songs)-1, loc=songs['dnce'].mean(), scale=st.sem(songs['dnce']))

(63.3097821732812, 65.44975348177684)

In [16]:
confidence = 0.99

st.t.interval(confidence, len(songs)-1, loc=songs['dnce'].mean(), scale=st.sem(songs['dnce']))

(62.97193258236186, 65.78760307269619)

In [17]:
confidence = 0.999

st.t.interval(confidence, len(songs)-1, loc=songs['dnce'].mean(), scale=st.sem(songs['dnce']))

(62.578166478431775, 66.18136917662629)

As you can see, the exact same applies as before.

What would happen to the interval with a smaller dataset? Let's take only the first 10% of the songs.

In [18]:
songs10Pct = songs.head(int(len(songs)*(10/100)))

In [22]:
confidence = 0.95

resultFullDataset = st.t.interval(confidence, len(songs)-1, loc=songs['dnce'].mean(), scale=st.sem(songs['dnce']))
resultFirst10PctDataset = st.t.interval(confidence, len(songs10Pct)-1, loc=songs10Pct['dnce'].mean(), scale=st.sem(songs10Pct['dnce']))

print('Interval with 95% confidence using the full dataset:', resultFullDataset)
print('Interval with 95% confidence using 10% of the dataset:', resultFirst10PctDataset)

Interval with 95% confidence using the full dataset: (63.3097821732812, 65.44975348177684)
Interval with 95% confidence using 10% of the dataset: (61.00511761637338, 68.12821571695994)


Takeaway: The smaller the dataset, the wider the range. This is because there is less data to draw conclusions from.